### PR TITLE
Refactors work with ranges

### PR DESCRIPTION
**Pull request by valeriyvan**

### CODE REVIEW

File: Common/Database/MockDatabase.swift [hunk 1]
```diff
@@ -31,10 +31,10 @@ public class FunctionCallTrace {
     public func diff(_ other: [String]) -> String {
         var diffs = [(String, String)]()
         log.enumerated().forEach { offset, logEntry in
-            if !(0..<other.count).contains(offset) {
-                diffs.append((logEntry, "MISSING"))
-            } else if logEntry != other[offset] {
+            if other.indices ~= offset {
```
Comment (DmitryBespalov) [inline Common/Database/MockDatabase.swift:34]: neat! I forgot this operator exists!

File: safe/safeUITests/Springboard.swift [hunk 2]
```diff
@@ -21,7 +21,7 @@ class Springboard {
         springboard.resolve()
 
         let safeIcons = springboard.icons.matching(identifier: "Safe")
-        for _ in (0..<safeIcons.count) {
+        for _ in 0..<safeIcons.count {
```
Comment (DmitryBespalov) [inline safe/safeUITests/Springboard.swift:24]: here also `safeIcons.indices`?

**Review by DmitryBespalov — APPROVED** [review]

File: safe/safeUITests/Springboard.swift (hunk 2 above)
Comment (valeriyvan) [reply safe/safeUITests/Springboard.swift:24]: No, it doesn't have indices.

File: Common/Database/MockDatabase.swift (hunk 1 above)
Comment (valeriyvan) [reply Common/Database/MockDatabase.swift:34]: That is why pair review is important!

**Pull request merged**